### ISSUE TITLE
customized objective function, getting error messages

### ISSUE THREAD

**Issue by fpishdadian** [tflearn/tflearn#235]
Hi, I am new to tensorflow and tflearn. I would like to define my own cost function. I have tried both methods mentioned in [issue #72 ](https://github.com/tflearn/tflearn/issues/72) but in both cases I get error messages. In the first case, the error message is:

from .config import _EPSILON, _FLOATX
ValueError: Attempted relative import in non-package

and in the second case:

"./tflearn/layers/estimator.py", line 135, in regression
exit()
NameError: global name 'exit' is not defined

I have spent some time trying to fix this, to no avail. Any help would be highly appreciated.

#########################################################################
Below is my code for the cost function:

```python
def deep_clustering(y_pred,y_true):
   
    with tf.name_scope("DeepClustering"):            
        n1,n2,c=y_true.shape
        N=n1*n2
        k=int(y_pred.shape[0]/N)
        
        Yc=tf.reshape(y_true,[N,c])   # target labels 
        Vk=tf.reshape(y_pred,[N,k])   # estimated embeddings
        
        YtY=tf.matmul(Yc.T,Yc)
        VtV=tf.matmul(Vk.T,Vk)  
        VtY=tf.matmul(Vk.T,Yc) 
        
        norm2Y_F=tf.trace(tf.matmul(YtY,YtY.T)) # squred Frobenius norm of YtY
        norm2V_F=tf.trace(tf.matmul(VtV,VtV.T)) # squared Frobenius norm of VtV
        norm2VY_F = tf.trace(tf.matmul(VtY,VtY.T)) # squared Frobenius norm of VtY
        
        loss = norm2Y_F - 2*norm2VY_F + norm2V_F        
    return loss
```

**Comment by aymericdamien** [comment]
Could you give the complete error trace? So that I can locate the error. (In latest TFLearn version, estimator.py line 135 suggest that there is an error with the metric).
Also,
```
from .config import _EPSILON, _FLOATX
```
Is it from your code? or from TFLearn library? If from your code, you should do from `tflearn.config import _EPSILON, _FLOATX`

**Comment by fpishdadian** [comment]
The line "from .config import ...." comes from TFLearn library. When I add my function to Objectives.py and run this (Objectives.py) script, it returns the following error:

Traceback (most recent call last):
  File "<ipython-input-158-b3c977dd15f7>", line 1, in <module>
    runfile('~/anaconda/lib/python2.7/site-packages/tflearn/objectives.py', wdir='~/anaconda/lib/python2.7/site-packages/tflearn')

  File "~/anaconda/lib/python2.7/site-packages/spyderlib/widgets/externalshell/sitecustomize.py", line 699, in runfile
    execfile(filename, namespace)

  File "~/anaconda/lib/python2.7/site-packages/spyderlib/widgets/externalshell/sitecustomize.py", line 81, in execfile
    builtins.execfile(filename, *where)

  File "~/anaconda/lib/python2.7/site-packages/tflearn/objectives.py", line 5, in <module>
    from .config import _EPSILON, _FLOATX

ValueError: Attempted relative import in non-package

############################################################################

When I pass the custom objective as a function inside the regression command, it returns:

'Tensor' object is not iterable.
Reminder: Custom loss function arguments must be define as follow: custom_loss(y_pred, y_true).
Traceback (most recent call last):

 File "~/anaconda/lib/python2.7/site-packages/spyderlib/widgets/externalshell/sitecustomize.py", line 699, in runfile
    execfile(filename, namespace)

  File "~/anaconda/lib/python2.7/site-packages/spyderlib/widgets/externalshell/sitecustomize.py", line 81, in execfile
    builtins.execfile(filename, *where)

  File "~/sample_code.py", line 86, in <module>
    network = regression(full_layer, optimizer='adam', loss=deep_clustering)

  File "~/anaconda/lib/python2.7/site-packages/tflearn/layers/estimator.py", line 135, in regression
    exit()

NameError: global name 'exit' is not defined

**Comment by aymericdamien** [comment]
When you pass your custom objective as a function, you should pass it as a function (without arguments):

```
regression(net, loss=my_custom_loss)
```

**Comment by fpishdadian** [comment]
Thanks for the quick replies!

I was calling the function correctly, exactly as you explained above. It turns out, however, that the problem was caused by the way the cost function was computed. I replaced .shape and .T and ... in the function definition by tensorflow math operations and it's working fine now. I am still wondering why it doesn't recognize the function when I add it to objective.py though.
